### PR TITLE
Initial `database_setup` RPC functions

### LINKED ISSUE #3654: database_setup.connect_existing

### LINKED ISSUE #3655: database_setup.create_new

### LINKED ISSUE #3656: databases.list

### PR DESCRIPTION
**Pull request by mathemancer**
Fixes #3654 
Fixes #3655 
Fixes #3656 

This adds `database_setup.create_new`, `database_setup.connect_existing`, and `databases.list` RPC functions.

**Technical details**

Note that while this PR doesn't actually break anything, it _does_ remove some functionality from the `connections` RPC functions. I think that's okay, since we'll be replacing those.

All RPC functions are documented, and the documentation should be sufficient to understand them. If not, we need to improve the docs before merging this PR.

**Screenshots**

## Checklist
 
- [X] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [X] My pull request targets the `develop` branch of the repository
- [X] My commit messages follow [best practices][best_practices].
- [X] My code follows the established code style of the repository.
- [X] I added tests for the changes I made (if applicable).
- [X] I added or updated documentation (if applicable).
- [X] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by pavish — APPROVED** [review]
Looks good to me!

File: mathesar/utils/permissions.py [hunk 1]
```diff
@@ -1,20 +1,118 @@
...
+    server, _ = Server.objects.get_or_create(host=host, port=port)
+    database, _ = Database.objects.get_or_create(
+        name=database_name, server=server
+    )
+    role, _ = Role.objects.get_or_create(
+        name=role_name,
+        server=server,
+        defaults={"password": password},
+    )
     return UserDatabaseRoleMap.objects.create(
```
Comment (Anish9901) [inline mathesar/utils/permissions.py:98]: This throws up an `IntegrityError` when we try to add the existing credentials via the API, I think we ought to use `get_or_create` here as well, to ensure that we allow users to reinstall mathesar even if the database gets deleted outside of mathesar.

**Review by Anish9901 — CHANGES_REQUESTED** [review]
Looks great overall @mathemancer, I've pointed out a minor bug please see the specific comment below.

File: mathesar/utils/permissions.py (hunk 1 above)
Comment (mathemancer) [reply mathesar/utils/permissions.py:98]: I'm okay with that change.

**Comment by mathemancer** [comment]
@Anish9901 Ready for re-review

**Review by Anish9901 — APPROVED** [review]
Looks great @mathemancer, Thanks!

**Pull request merged**